docs/layer_util.js

```diff
@@ -324,20 +323,58 @@ function createTileCallback(layerDisplayData, resolve) {
  * Displays a collection of tiles (given by the 'urls' attribute of layer)
  * on the map using a CompositeImageMapType to combine all tiles for a given
  * map location into one tile.
+ *
+ * The urls can be either raw urls or .json urls in the form provided by NOAA,
+ * like
+ * https://storms.ngs.noaa.gov/storms/tilesd/services/tileserver.php?/20170827-rgb.json.
+ * In the latter case, the layerDisplayData.overlay object will only be created
+ * once all JSONs are downloaded and their corresponding tile urls are
+ * retrieved.
+ *
+ * TODO(janakr): JSON files can tell us map bounds and zoom levels.
  * @param {google.maps.Map} map
  * @param {Object} layer Data for layer coming from Firestore
  * @return {Promise} Promise that resolves when images are all downloaded
  */
 function addTileLayer(map, layer) {
   const layerDisplayData = new LayerDisplayData(null, true);
   layerArray[layer['index']] = layerDisplayData;
-  layerDisplayData.overlay = new CompositeImageMapType({
-    tileUrls: layer['urls'],
-    tileSize: layer['tile-size'],
-    maxZoom: layer['maxZoom'],
-    opacity: layer['opacity'],
+  const urlPromises = [];
+  for (const url of layer['urls']) {
+    if (url.match(/{Z}/i) && url.match(/{Y}/i) && url.match(/{X}/i)) {
```

Comment:
can you do some commenting of regex expressions for the uninitiated

Reply:
Done, here and in CompositeImageMapType class.